[Go] feat: update docs for dd-trace-go/v2 (#25897)
DataDog/documentation: content/en/profiler/enabling/go.md

2. Get `dd-trace-go` using the command:

    ```shell

>     go get gopkg.in/DataDog/dd-trace-go.v1/profiler // v1
>     // go get github.com/DataDog/dd-trace-go/v2/profiler // v2

Reviewer: ```suggestion
    go get gopkg.in/DataDog/dd-trace-go.v1/profiler # v1
    # go get github.com/DataDog/dd-trace-go/v2/profiler # v2
```

    ```
     **Note**: Profiler is available in the `dd-trace-go` library for versions 1.23.0+.

3. Import the [profiler][6] ([or profiler v2][21]) at the start of your application:

>     ```Go

Reviewer: ```suggestion
    ```go
```

    ```shell

>     go get gopkg.in/DataDog/dd-trace-go.v1/profiler

Reviewer: ```suggestion
    go get gopkg.in/DataDog/dd-trace-go.v1/profiler // v1
    // go get github.com/DataDog/dd-trace-go/v2/profiler // v2
```

    ```

    If you are using the preview version of 2.x, use:

    ```shell
    go get github.com/DataDog/dd-trace-go/v2/profiler
    ```

     **Note**: Profiler is available in the `dd-trace-go` library for versions 1.23.0+.

> 3. Import the [profiler][6] ([v2 documentation][21]) at the start of your application:

Reviewer: ```suggestion
3. Import the [profiler][6] ([or profiler v2][21]) at the start of your application:
```

    ```Go

>     import "gopkg.in/DataDog/dd-trace-go.v1/profiler" // use "github.com/DataDog/dd-trace-go/v2/profiler" if you're using 2.x

Reviewer: ```suggestion
    import "gopkg.in/DataDog/dd-trace-go.v1/profiler" // 1.x
    // "github.com/DataDog/dd-trace-go/v2/profiler" // 2.x
```

6. After a minute or two, visualize your profiles in the [Datadog APM > Profiler page][10].

> **Note**: By default, only the CPU and Heap profiles are enabled. Use [profiler.WithProfileTypes][11] to enable additional [profile types][12]. For relevant v2 documentation, view [profiler.WithProfileTypes][22] and [profile types][23].

Reviewer: ```suggestion
**Note**: By default, only the CPU and Heap profiles are enabled. Use [profiler.WithProfileTypes][11] to enable additional [profile types][12]. For go v2, view [profiler.WithProfileTypes][22] and [profile types][23].
```

Reviewer: The suggestion I put here isn't much better and it's because it's fairly difficult to link out to external references for v1 and v2 in a concise way without making v2 seem second class

To improve this, I recommend having a page that describes dd-trace-go with a `v1` and `v2` tab that allow customers to see only what's relevant for them and have this just link there. It's an extra click to get to `profiler.WithProfileTypes` or similar, but it reduces the confusion/clutter within this doc and many others
